### PR TITLE
📖 Update documentation with latest changes

### PR DESCRIPTION
**Pull request by furkatgofurov7**
This PR updates CI badges for master branch of IPAM with v1a5 links and fixes some minor issues in the api documentation accordingly to IPAM objects with all required fields needed to run properly example generation Makefile targets.

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
/retitle 📖 Update documentation with latest changes

**Review by kashifest — APPROVED** [review]

File: docs/api.md [hunk 1]
```diff
@@ -34,6 +36,7 @@ The *spec* field contains the following :
 
 * **clusterName**: That is the name of the cluster to which this pool belongs
   it is used to verify whether the resource is paused.
+* **namePrefix**: That is is the prefix used to generate the IPAddress.
```
Comment (fmuyassarov) [inline docs/api.md:39]:
```suggestion
* **namePrefix**: That is the prefix used to generate the IPAddress.
```

File: docs/api.md [hunk 2]
```diff
@@ -80,11 +85,14 @@ apiVersion: ipam.metal3.io/v1alpha1
...
   name: pool1-192-168-0-13
+  namespace: default
 spec:
   pool:
-    Name: pool1
+    name: pool1
+    namespace: default
   claim:
-    Name: claim1
+    name: claim1
```
Comment (fmuyassarov) [inline docs/api.md:94]: you can probably update this name here as I mentioned in https://github.com/metal3-io/ip-address-manager/pull/41#discussion_r678994631

Comment (furkatgofurov7) [reply docs/api.md:94]: This is taken care in #41

File: docs/api.md (hunk 1 above)
Comment (furkatgofurov7) [reply docs/api.md]: Good catch, thanks for review/

File: docs/api.md (hunk 2 above)
Comment (furkatgofurov7) [reply docs/api.md]: Sorry, disregard above comment. I updated document in this PR and 41 is only for examples folder related changes.

**Review by fmuyassarov — COMMENTED** [review]
Looks good. Can you also update _Compatibility with Cluster API_ section in README to include new API versions ?

**Comment by furkatgofurov7** [comment]
>  Can you also update _Compatibility with Cluster API_ section in README to include new API versions ?

I am planning to do it after we release IPAM from master branch. Reason is, we do not have a IPAM release that supports CAPI v1a4 now and it is confusing to put here as IPAM release `v0.1.X`, once we release I will do something like:
```
| IPAM version      | CAPM3 version     | Cluster API version | IPAM Release |
|-------------------|-------------------|---------------------|--------------|
| v1alpha1 (v0.1.X) | v1alpha4 (v0.4.X) | v1alpha3 (v0.3.X)   | v0.0.X       |
| v1alpha1 (v0.1.X) | v1alpha5 (v0.5.X) | v1alpha4 (v0.4.X)   | v0.1.X       |

```

**Comment by fmuyassarov** [comment]
> > Can you also update _Compatibility with Cluster API_ section in README to include new API versions ?
> 
> I am planning to do it after we release IPAM from master branch. Reason is, we do not have a IPAM release that supports CAPI v1a4 now and it is confusing to put here as IPAM release `v0.1.X`, once we release I will do something like:
> 
> ```
> | IPAM version      | CAPM3 version     | Cluster API version | IPAM Release |
> |-------------------|-------------------|---------------------|--------------|
> | v1alpha1 (v0.1.X) | v1alpha4 (v0.4.X) | v1alpha3 (v0.3.X)   | v0.0.X       |
> | v1alpha1 (v0.1.X) | v1alpha5 (v0.5.X) | v1alpha4 (v0.4.X)   | v0.1.X       |
> ```

Ah right, I forgot that we haven't yet released. +1
/lgtm

**Pull request merged**